Gems/ROS2/gem.json

```diff
@@ -1,6 +1,6 @@
 {
     "gem_name": "ROS2",
-    "version": "3.1.0",
+    "version": "3.2.1",
```

Comment:
Why was this version changed? It is now ahead of the `development` branch. It also means that we bumped the version number from `2.0.0` up to `3.2.1` before the two stable releases, which I do not understand.

Reply:
The current release package for the ROS2 gem was version 3.2.0. For some reason, this version update wasn't cherry picked back to development (which was at 3.1.0), and we need the minor version bump to update the released gem zip file. 
As for 2.0.0 from the previous stable version, this was aa major version bump to support the PhysX split due to the gem dependency changes for related gems and templates (It needed to change PhysX -> PhysX5) so all of their versions needed a bump as well. (see https://github.com/o3de/ROSConDemo/pull/276 )